### DISCUSSION TITLE
Allow using mods in the editor

### DISCUSSION THREAD

**Discussion by Avelia-tr** [ppy/osu#26321] [Ideas]
With the implementation of so many mods that changes how the game is displayed/played in so many way, it would be really cool to be able to map with those mods enable. That would allow to make maps that fits certain mods with less difficulty (ie maps that fits some configuration of barrel roll)
Here is an example of someone asking for it and showing what the effect could look like in the editor https://twitter.com/minetoblend/status/1741977441273319646?s=20

I know there already was a discussion about something near it ( https://github.com/ppy/osu/discussions/17396 ) but it was more about playtesting, here it's for a more generalised purpose in the editor. (obviously both would be really cool to add)

**Comment by minetoblend** [comment]
Hi, I'm the person that made the video from the tweet linked above (Wanted to make a discussion about this myself just now but looks like someone was quicker than me). There are a handful of mods that I think would be cool to be able to preview in the editor, mainly to make it easier to make maps that are meant to be played with certain mods/mod combinations, but also be able to see the map in the way it's target audience is most likely going to play the map.
For a mod to make sense to display in the editor it needs to meet 2 requirements imo:
- it visually changes the playfield/hitobjects.
- it needs to be deterministic (stuff like the magnetised mod that depends on cursor position wouldn't work for obvious reasons).

I've split the mods into 2 categories based on what purpose I think supporting them would have.

### Mods for gimmick maps
I think these mods could be used to create interesting/unique gameplay if the right person with the right tools were to make a map for them.

- Barrel Roll
- Depth
- Traceable
- Spin In
- Grow/Deflate
- Wind Down/Up 
- Approach Different
- Bubbles
- Flashlight (Hiding 3/4 of the playfield would probably cause more issues than it solves though so this one sounds pretty unpractical)

### General/other mods
I don't really see those two as anything you'd make a gimmick map for, but I think since they're the mods that a lot of players are most likely to use it would still make sense to be able to preview them in the editor.
- Hidden
- Double time (Adding 150% option to playback speed would also work here)

As for the other game modes, I have little to no knowledge about what mods there are for them, so I won't attempt to compile a list of them.

## Adding ability to store "recommended" mods on beatmaps
Currently gimmick maps that are meant to be played with specific mods typically use the background/storyboard/description to tell players how they're supposed to be played. Since mods can be customised in lazer, telling players to use any mod combination with non-standard settings in that way is gonna be a frustrating experience.
I think giving the mapper the ability to specify a "recommended" mod combination on a beatmap would greatly benefit these maps. 
I'm not 100% sure what the best way to handle UX for that from the player's perspective. I'm thinking about either showing a prompt to the player (should only prompt the first time the map is being played, would become annoying otherwise), or maybe showing a little popup next to the map in song select, letting the player know about the suggested mods & a with button to apply them.

**Comment by iamateacup** [comment]
Was going to make a feature request but saw this was already here. Having all mods wasnt even what i was thinking about, but the main thing would be a 150% playback speed in the editor. A hidden mod viewer would also maybe be nice, but I understand that this doesnt really fit in the current editor, ie it would need a whole new mod section. A 150% playback section would be super useful though for both mappers and players that want to preview maps
